### PR TITLE
Fix docstrings of get_latest_versions

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Fix docstrings of get_latest_versions

## How is this patch tested?

Manually verified the doc is built properly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@mparkhe Does `ModelVersionDetailed` still exist?

File: mlflow/tracking/_model_registry/client.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def get_latest_versions(self, name, stages=None):
         :param name: Name of the registered model to update.
         :param stages: List of desired stages. If input list is None, return latest versions for
                        for 'Staging' and 'Production' stages.
-        :return: List of `:py:class:`mlflow.entities.model_registry.ModelVersionDetailed` objects.
```
Comment (mparkhe) [inline mlflow/tracking/_model_registry/client.py]: That is good catch! Thank you.

**Review by mparkhe — APPROVED** [review]

File: mlflow/tracking/_model_registry/client.py (hunk 1 above)
Comment (harupy) [reply mlflow/tracking/_model_registry/client.py]: Some files still contain `ModelVersionDetailed`. I will replace them with `ModelVersion`.

https://github.com/mlflow/mlflow/search?q=ModelVersionDetailed&unscoped_q=ModelVersionDetailed

Comment (harupy) [reply mlflow/tracking/_model_registry/client.py]: Can I remove this section?

https://github.com/mlflow/mlflow/blame/eea0843e7bdcbcba59cd6d3cfb84de565f40071b/docs/source/rest-api.rst#L1937

Comment (mparkhe) [reply mlflow/tracking/_model_registry/client.py]: Don't worry about the documentation on `REST APIs` page. That will be autogenerated as a part of the release process.

**Comment by harupy** [comment]
Some files still contain `ModelVersionDetailed`.

https://github.com/mlflow/mlflow/search?q=ModelVersionDetailed&unscoped_q=ModelVersionDetailed

**Comment by mparkhe** [comment]
> Some files still contain `ModelVersionDetailed`.
> 
> https://github.com/mlflow/mlflow/search?q=ModelVersionDetailed&unscoped_q=ModelVersionDetailed

Thanks for the pointer. I will take care of this.

**Pull request merged**